ASTE Turbine Tutorial (#244)
precice/tutorials: aste-turbine/README.md

---
title: ASTE (Artificial Solver Testing Environment) Turbine Tutorial
permalink: tutorials-aste-turbine.html
keywords: ASTE, Testing, Turbine
summary: This tutorial is an example case for ASTE, showcasing basic features and usage of ASTE.
---

{% include note.html content="Get the [case files of this tutorial](https://github.com/precice/tutorials/tree/master/aste-turbine). Read how in the [tutorials introduction](https://precice.org/tutorials.html)." %}

## Setup

This case is a simple usage scenario for ASTE. Some features demonstrated by this tutorial:

* Mapping from a fine grid (`0.009.vtk`) to a coarse grid (`0.01.vtk`).
* Usage of ASTE partitioner.
* Usage of ASTE mesh joiner.

> * Usage of ASTE Calculator for calculating a function on given mesh.

Reviewer: ```suggestion
* Usage of ASTE calculator for calculating a function on a given mesh.
```

* Usage of ASTE.

> ## Running the ASTE

Reviewer: ```suggestion
## Running ASTE
```

Run the `run.sh` script. It performs the following steps:

* Downloads the meshes from preCICE repository.
* Using `vtk_calculator.py` script, calculates function `x^2+exp(y^3)+cos(z)` on a finer grid.
* Using `partition_mesh.py` script, partitions the coarse and fine mesh into 2 domains.
* Using `preciceMap` executable, maps the data from the fine grid to the coarse grid.
* Using `join_mesh.py`script, joins the result meshes into a final mesh.
* Using ``vtk_calculator.py` script, calculates difference between mapped and original function on coarse grid.

> Important note:  The `run.sh` script assumes that the ASTE binaries and python scripts are in `$PATH` or ASTE is installed.

Reviewer: Maybe give a quick check for this and link to the new ASTE documentation page. This is currently not mentioned anywhere here.

# ---
title: ASTE (Artificial Solver Testing Environment) Turbine Tutorial

> permalink: tutorials-aste-turbine.html

Author: Thank you for notice, here : https://github.com/precice/precice.github.io/pull/104

keywords: ASTE, Testing, Turbine
summary: This tutorial is an example case for ASTE, showcasing basic features and usage of ASTE.
---

{% include note.html content="Get the [case files of this tutorial](https://github.com/precice/tutorials/tree/master/aste-turbine). Read how in the [tutorials introduction](https://precice.org/tutorials.html)." %}

## Setup

> This case is a simple usage scenario for. Some features offered by this case:

Reviewer: oops: for what? :sweat_smile: 

I guess you meant to link to ASTE and some documentation here.

Also, maybe write "features demonstrated" instead of "features offered".

* Mapping from a fine grid (`0.009.vtk`) to a coarse grid (`0.01.vtk`).
* Usage of ASTE partitioner.
* Usage of ASTE mesh joiner.
* Usage of ASTE Calculator for calculating a function on given mesh.
* Usage of ASTE.

## Running the ASTE

Run the `run.sh` script. It performs the following steps:

* Downloads the meshes from preCICE repository.

> * Using `vtk_calculator.py` script, calculates function `x + y` on a finer grid.

Reviewer 2: Let's try to use a more sophisticated test function here. There are some standard functions which are often used to test RBF interpolation, e.g the [eggholder function](https://www.sfu.ca/~ssurjano/egg.html).

Author: Thank you, I will definitely consider using this.

* Partitions the coarse and fine mesh into 2 domains.
* Map the data from fine grid to coarse grid.
* Join the result meshes into a final mesh.

> * Using calculator calculates difference between mapped and original function on coarse grid.

Reviewer: Maybe add a few more details on each step?

Author: I added the scripts which are used for each step.

> * Map the data from fine grid to coarse grid.
> * Join the result meshes into a final mesh.
> * Using calculator calculates difference between mapped and original function on coarse grid.

Reviewer: Notice how I change the beginning of each bullet point to follow the same style:
```suggestion
* Maps the data from fine grid to coarse grid.
* Joins the result meshes into a final mesh.
* Calculates difference between mapped and original function on coarse grid.
```

Author: Small detail makes huge difference :+1:

> Important note:  The `run.sh` script assumes that the ASTE binaries and python scripts are in `$PATH`.

Reviewer 3: Does it work for you when you use (from `PATH` in a different directory) `partition_mesh.py --algorithm topology` which relies on `libmetis`. Also, it would be good to have an install target in the `CMakeLists.txt` of `ASTE` so that user can easily install it without prepending system paths.

Author: You are right it is not working with `libmetis`, since in python script look `metisAPI.so` in `cwd`. 

Now it is fixed in https://github.com/precice/aste/pull/53. 

I opened an issue about installation
https://github.com/precice/aste/issues/54
